src/Compilers/CSharp/Portable/BoundTree/OutDeconstructVarPendingInference.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class OutDeconstructVarPendingInference
+    {
+        public BoundDeconstructValuePlaceholder Placeholder;
+
+        public BoundDeconstructValuePlaceholder SetInferredType(TypeSymbol type, bool success)
```

Comment:
This is implementing an essentially mutable bound node. Why can this happen only once? What happens during overload resolution?